src/Symfony/Component/BrowserKit/AbstractBrowser.php

```diff
@@ -356,7 +356,8 @@ public function request(string $method, string $uri, array $parameters = [], arr
 
         $server['HTTPS'] = 'https' == parse_url($uri, \PHP_URL_SCHEME);
 
-        $this->internalRequest = new Request($uri, $method, $parameters, $files, $this->cookieJar->allValues($uri), $server, $content);
+        $requestFqcn = $this->getRequestClass();
+        $this->internalRequest = new $requestFqcn($uri, $method, $parameters, $files, $this->cookieJar->allValues($uri), $server, $content);
```

Comment:
you cant assume the constructor signature :)

Reply:
You're absolutely right! I changed it for a factory which makes it's just a bit verbose but fixes this issue